hail/python/hailtop/auth/sql_config.py

```diff
@@ -1,22 +1,94 @@
+from typing import Dict, NamedTuple, Optional, Any
 import json
+import os
 
 
-def create_secret_data_from_config(config, server_ca, client_cert, client_key):
-    assert all(config.get(x) is not None
-               for x in ('host', 'user', 'password', 'db', 'ssl-ca', 'ssl-cert',
-                         'ssl-key', 'ssl-mode')), config.keys()
-    secret_data = dict()
-    secret_data['sql-config.json'] = json.dumps(config)
-    secret_data['sql-config.cnf'] = f'''[client]
-host={config["host"]}
-user={config["user"]}
-password="{config["password"]}"
-database={config["db"]}
-ssl-ca={config["ssl-ca"]}
-ssl-cert={config["ssl-cert"]}
-ssl-key={config["ssl-key"]}
-ssl-mode={config["ssl-mode"]}
+class SQLConfig(NamedTuple):
+    host: str
+    port: int
+    user: str
+    password: str
+    instance: str
+    connection_name: str
```

Comment:
Remind me -- is this new syntax going to force an upgrade of the Hail version required by users?

Reply:
Nah, it's in 3.6: https://docs.python.org/3.6/library/typing.html#typing.NamedTuple. It's not in 3.5: https://docs.python.org/3.5/library/typing.html#typing.NamedTuple.

The issue from before was my use of `re.Pattern` vs `typing.Pattern`.

But good looking out for version incompatibilities!